lib/shoryuken/queue.rb

```diff
@@ -44,7 +45,8 @@ def receive_messages(options)
     end
 
     def fifo?
-      @_fifo ||= queue_attributes.attributes[FIFO_ATTR] == 'true'
+      return @_fifo unless @_fifo == :uninitialized
```

Comment:
Hi @timbreitkreutz 

Thanks for the explanation [here](https://github.com/phstc/shoryuken/pull/528#discussion_r229138318). 

How your previous solution with `defined?`?

```suggestion
def fifo?
  # Make sure the memoization work with boolean to avoid multiple calls to SQS
  # see https://github.com/phstc/shoryuken/pull/529
  return @_fifo if defined?(@_fifo)
  @_fifo = queue_attributes.attributes[FIFO_ATTR] == 'true'
end
```

Reply:
PS1: If you changed it because of reek, don't worry about it, I think `defined?` is fine.
PS2: Great job on the fix! Thank you 🍻

Reply:
yeah reek didn't like it that way.